apm-agent-core/src/main/java/co/elastic/apm/agent/impl/stacktrace/StacktraceConfiguration.java

```diff
@@ -88,7 +104,15 @@ public int getStackTraceLimit() {
         return stackTraceLimit.get();
     }
 
-    public long getSpanFramesMinDurationMs() {
-        return spanFramesMinDurationMs.getValue().getMillis();
+    public long getSpanStackTraceMinDurationMs() {
+        if (!spanFramesMinDurationMs.isDefault()) {
```

Comment:
My understanding of what we want is that

1. if spanStackTraceMinDurationMs is **not** default, it takes precedence and no need to check spanFramesMinDurationMs at all
2. if spanStackTraceMinDurationMs **is** default, then look at spanFramesMinDurationMs and if it is **not** default then do this 
The code logic doesn't reflect that

@felixbarny please confirm or refute, in case I got the intended logic wrong

Reply:
That's correct.

Reply:
fixed logic